File: leetcode-04/solutions/cpp/0209.cpp
class Solution {
 public:
  int minSubArrayLen(int s, vector<int>& nums) {
    int ans = INT_MAX;
    int sum = 0;
    int j = 0;

    for (int i = 0; i < nums.size(); i++) {
      sum += nums[i];
      while (sum >= s) {
        ans = min(ans, i - j + 1);
        sum -= nums[j++];
      }
    }

    return ans == INT_MAX ? 0 : ans;
  }
};

File: leetcode-04/solutions/python3/0209.py
class Solution:
    def minSubArrayLen(self, s: int, nums: List[int]) -> int:
        ans = float('inf')
        sum = 0
        j = 0

        for i in range(len(nums)):
            sum += nums[i]
            while sum >= s:
                ans = min(ans, i - j + 1)
                sum -= nums[j]
                j += 1

        return 0 if ans == float('inf') else ans
